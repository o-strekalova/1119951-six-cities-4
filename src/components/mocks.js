export const offersAll = [{
  id: `1010`,
  city: {
    location: {
      lat: 52.3909553943508,
      long: 4.85309666406198,
      zoom: 13,
    },
    name: `Amsterdam`,
  },
  location: {
    lat: 52.3909553943508,
    long: 4.85309666406198,
    zoom: 13,
  },
  pictures: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  preview: `img/apartment-01.jpg`,
  title: `Wood and stone place`,
  type: `room`,
  price: 999,
  isPremium: true,
  isFavorite: false,
  rating: 4.4,
  description: ``,
  bedrooms: 2,
  guests: 4,
  features: [],
  owner: {
    avatar: `https://api.adorable.io/avatars/128`,
    name: ``,
    isSuper: true,
    id: 111,
  },
}, {
  id: `2020`,
  city: {
    location: {
      lat: 52.3909553943508,
      long: 4.85309666406198,
      zoom: 13,
    },
    name: `Paris`,
  },
  location: {
    lat: 52.3909553943508,
    long: 4.85309666406198,
    zoom: 13,
  },
  pictures: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  preview: `img/apartment-01.jpg`,
  title: `Wood and stone place`,
  type: `room`,
  price: 999,
  isFavorite: false,
  isPremium: true,
  rating: 4.4,
  description: ``,
  bedrooms: 2,
  guests: 4,
  features: [],
  owner: {
    avatar: `https://api.adorable.io/avatars/128`,
    name: ``,
    isSuper: true,
    id: 222,
  },
}];

export const offersRaw = [{
  "id": 1010,
  "city": {
    "location": {
      "latitude": 52.3909553943508,
      "longitude": 4.85309666406198,
      "zoom": 13,
    },
    "name": `Amsterdam`,
  },
  "location": {
    "latitude": 52.3909553943508,
    "longitude": 4.85309666406198,
    "zoom": 13,
  },
  "images": [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  "preview_image": `img/apartment-01.jpg`,
  "title": `Wood and stone place`,
  "type": `room`,
  "price": 999,
  "is_premium": true,
  "is_favorite": false,
  "rating": 4.4,
  "description": ``,
  "bedrooms": 2,
  "max_adults": 4,
  "goods": [],
  "host": {
    "avatar_url": `https://api.adorable.io/avatars/128`,
    "name": ``,
    "is_pro": true,
    "id": 111,
  },
}, {
  "id": 2020,
  "city": {
    "location": {
      "latitude": 52.3909553943508,
      "longitude": 4.85309666406198,
      "zoom": 13,
    },
    "name": `Paris`,
  },
  "location": {
    "latitude": 52.3909553943508,
    "longitude": 4.85309666406198,
    "zoom": 13,
  },
  "images": [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  "preview_image": `img/apartment-01.jpg`,
  "title": `Wood and stone place`,
  "type": `room`,
  "price": 999,
  "is_favorite": false,
  "is_premium": true,
  "rating": 4.4,
  "description": ``,
  "bedrooms": 2,
  "max_adults": 4,
  "goods": [],
  "host": {
    "avatar_url": `https://api.adorable.io/avatars/128`,
    "name": ``,
    "is_pro": true,
    "id": 222,
  },
}];

export const reviews = [{
  id: 1110,
  avatar: `https://api.adorable.io/avatars/128`,
  name: `Angelina`,
  rating: 4.4,
  date: new Date(2020, 10, 5, 10),
  text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
}];

export const authInfo = {
  avatar: `https://api.adorable.io/avatars/128`,
  email: `mail@mail.ru`,
  id: 1,
  isSuper: true,
  name: `Angelina`,
};
