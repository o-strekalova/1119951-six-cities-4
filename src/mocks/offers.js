const OFFERS_TITLES = [`Beautiful & luxurious apartment at great location`, `Wood and stone place`, `Canal View Prinsengracht`, `Nice, cozy, warm big bed apartment`];
const OFFERS_TYPES = [`apartment`, `room`, `house`, `hotel`];
const AVATAR_URL = `https://api.adorable.io/avatars/128`;
const COORDINATES = [
  [52.3909553943508, 4.85309666406198],
  [52.369553943508, 4.85309666406198],
  [52.3909553943508, 4.929309666406198],
  [52.3809553943508, 4.939309666406198]
];
const OFFERS_COUNT = 4;

const getRandomIntegerNumber = (min, max) => {
  return min + Math.floor(Math.random() * (max - min));
};

const getRandomArrayItem = (array) => {
  const randomIndex = getRandomIntegerNumber(0, array.length);
  return array[randomIndex];
};

const generateOffers = () => {
  const offers = [];

  for (let i = 0; i < OFFERS_COUNT; i++) {
    offers.push({
      id: i,
      coords: COORDINATES[i],
      pictures: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
      title: getRandomArrayItem(OFFERS_TITLES),
      type: getRandomArrayItem(OFFERS_TYPES),
      price: getRandomIntegerNumber(50, 1000),
      isPremium: Math.random() > 0.5,
      rating: getRandomIntegerNumber(0, 4) + Math.round(Math.random() * 10) / 10,
      description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
      bedrooms: getRandomIntegerNumber(1, 10),
      guests: getRandomIntegerNumber(1, 20),
      features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
      owner: {
        avatar: AVATAR_URL,
        name: `Angelina`,
        isSuper: Math.random() > 0.5,
      },
    });
  }

  return offers;
};

export const offers = generateOffers();
