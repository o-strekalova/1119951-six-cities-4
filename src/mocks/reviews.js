const AVATAR_URL = `https://api.adorable.io/avatars/128`;
const REVIEWS_COUNT = 11;

const getRandomIntegerNumber = (min, max) => {
  return min + Math.floor(Math.random() * (max - min));
};

const generateReviews = () => {
  const reviews = [];

  for (let i = 0; i < REVIEWS_COUNT; i++) {
    reviews.push({
      id: i,
      avatar: AVATAR_URL,
      name: `Angelina`,
      rating: getRandomIntegerNumber(0, 4) + Math.round(Math.random() * 10) / 10,
      date: new Date(),
      text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
    });
  }

  return reviews;
};

export const reviews = generateReviews();
