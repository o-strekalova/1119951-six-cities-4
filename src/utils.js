export const CardsListClass = {
  MAIN: `cities__places-list tabs__content`,
  PROPERTY: `near-places__list`,
};

export const CardClass = {
  MAIN: `cities__place-card`,
  PROPERTY: `near-places__card`,
};

export const SortType = {
  POPULAR: `Popular`,
  PRICE_FROM_LOW: `Price: low to high`,
  PRICE_FROM_HIGH: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getRatingPercentage = (rating) => {
  return Math.round(rating) * 20;
};
