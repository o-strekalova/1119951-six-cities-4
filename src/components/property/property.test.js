import React from "react";
import renderer from "react-test-renderer";
import Property from "./property.jsx";
import {Router} from "react-router-dom";
import {offersAll, authInfo, reviews} from "../mocks";
import {AuthorizationStatus, noop} from "../../utils";
import history from "../../history";

it(`Render Property for authorized user`, () => {
  const tree = renderer
    .create(
        <Router
          history={history}
        >
          <Property
            authInfo={authInfo}
            authorizationStatus={AuthorizationStatus.AUTH}
            errorMessage={null}
            isToggleChecked={offersAll[0].isFavorite}
            offer={offersAll[0]}
            offersNear={offersAll}
            reviews={reviews}
            onCardTitleClick={noop}
            onFavoriteButtonClick={noop}
            onLogoClick={noop}
            onReviewSubmit={noop}
            onToggleClick={noop}
            onUserNameClick={noop}
          />
        </Router>, {
          createNodeMock: () => {
            return document.createElement(`DIV`);
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Property for unauthorized user`, () => {
  const tree = renderer
    .create(
        <Router
          history={history}
        >
          <Property
            authInfo={{}}
            authorizationStatus={AuthorizationStatus.NO_AUTH}
            errorMessage={null}
            isToggleChecked={offersAll[0].isFavorite}
            offer={offersAll[0]}
            offersNear={offersAll}
            reviews={reviews}
            onCardTitleClick={noop}
            onFavoriteButtonClick={noop}
            onLogoClick={noop}
            onReviewSubmit={noop}
            onToggleClick={noop}
            onUserNameClick={noop}
          />
        </Router>, {
          createNodeMock: () => {
            return document.createElement(`DIV`);
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Property with ErrorMessage`, () => {
  const tree = renderer
    .create(
        <Router
          history={history}
        >
          <Property
            authInfo={authInfo}
            authorizationStatus={AuthorizationStatus.AUTH}
            errorMessage={`Failed to post review`}
            isToggleChecked={offersAll[0].isFavorite}
            offer={offersAll[0]}
            offersNear={offersAll}
            reviews={reviews}
            onCardTitleClick={noop}
            onFavoriteButtonClick={noop}
            onLogoClick={noop}
            onReviewSubmit={noop}
            onToggleClick={noop}
            onUserNameClick={noop}
          />
        </Router>, {
          createNodeMock: () => {
            return document.createElement(`DIV`);
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
