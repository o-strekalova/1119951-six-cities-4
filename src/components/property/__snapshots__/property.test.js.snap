// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Property 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Wood and stone place"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Wood and stone place"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Wood and stone place"
              className="property__image"
              src="img/apartment-03.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Wood and stone place"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Wood and stone place"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Wood and stone place"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Wood and stone place
            </h1>
            <button
              className="property__bookmark-button button"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.4
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Room
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              2
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              4
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              999
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            />
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="https://api.adorable.io/avatars/128"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                1
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="https://api.adorable.io/avatars/128"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Angelina
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-10-5"
                  >
                     
                    November 2020
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              className="reviews__form form"
              method="post"
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  className="form__rating-input visually-hidden"
                  id="5-stars"
                  name="rating"
                  type="radio"
                  value={5}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="4-stars"
                  name="rating"
                  type="radio"
                  value={4}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="good"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="3-stars"
                  name="rating"
                  type="radio"
                  value={3}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="not bad"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="2-stars"
                  name="rating"
                  type="radio"
                  value={2}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="badly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="1-stars"
                  name="rating"
                  type="radio"
                  value={1}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-stars"
                  title="terribly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                id="review"
                name="review"
                placeholder="Tell how was your stay, what you like and what can be improved"
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled=""
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <div
          id="map"
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
            onMouseOver={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Wood and stone place"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-01.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    999
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
                onClick={[Function]}
              >
                <a
                  href="#"
                >
                  Wood and stone place
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Room
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
